OpenEdXMobile/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -69,52 +80,118 @@ private void createAssessmentTable(SQLiteDatabase db){
+                                        "^" + TextUtils.join("/", Arrays.<CharSequence>asList(
+                                                previousAppDirPath, username)),
+                                        TextUtils.join("/", Arrays.<CharSequence>asList(
+                                                externalAppDir.getAbsolutePath(),
+                                                AppConstants.Directories.VIDEOS,
+                                                hashedUsername)).toString());
+
+                                // First update the name and path of the videos directory
+                                final File previousDir = new File(externalAppDir, username);
+                                if (previousDir.exists()) {
+                                    final File newDir = new File(externalAppDir,
+                                            TextUtils.join("/", Arrays.<CharSequence>asList(
+                                                    AppConstants.Directories.VIDEOS,
+                                                    hashedUsername)).toString());
+                                    if (!((newDir.mkdirs() || newDir.exists()) &&
+                                            previousDir.renameTo(newDir))) {
+                                        continue;
+                                    }
+                                }
+
+                                // Then update the database row
+                                final ContentValues updatedValues = new ContentValues();
+                                updatedValues.put(DbStructure.Column.USERNAME, hashedUsername);
+                                updatedValues.put(DbStructure.Column.FILEPATH, newFilePath);
+                                db.update(DbStructure.Table.DOWNLOADS, updatedValues,
+                                        DbStructure.Column.ID + "= ?", new String[]{id});
+                            }
+                        } finally {
+                            cursor.close();
+                        }
+
+                        // Now migrate the subtitles directory
+                        final File previousSrtDir = new File(externalAppDir, "srtFolder");
+                        if (previousSrtDir.exists()) {
+                            final File newSrtDir = new File(externalAppDir,
+                                    AppConstants.Directories.VIDEOS + "/" + AppConstants.Directories.SUBTITLES);
+                            newSrtDir.mkdirs();
+                            previousSrtDir.renameTo(newSrtDir);
+                        }
+                    }
+                }
 
-            logger.debug("Database upgraded from " + oldVersion + " to " + newVersion);
-        }catch(Exception e){
-            logger.error(e);
+                logger.debug("Database upgraded from " + oldVersion + " to " + newVersion);
+                db.setTransactionSuccessful();
```

Comment:
You're [still][1] marking it as successful unconditionally. This should be moved to inside the `if (externalAppDir != null)` condition at the end (i.e. just 2-3 lines above before ending it.

[1]: https://github.com/edx/edx-app-android/pull/797#discussion_r81094549